# Exploring the ADNI dataset using Pandas

Dataset from http://adni.loni.usc.edu. This is a bit of a tutorial on exploring the data with Pandas.

First, how to load matrices in to a Pandas data frame and start exploring them. The code (module) below was put together by Pablo, and is the start of a package to deal with the ADNI data in Python that we can make as a group. This is called 'load_ADNI'. 

We need to load some packages (and modules). With the standard Anaconda distribution I think most of the packages will already be included. However, if you are ever missing anything, the easiest way to get it is to 'pip install ...' on the command line. 

In [1]:
from scipy.io import loadmat #scipy for dealing with scientific specific things -- eg. matrices from matlab
import os #dealing with features of the operating system you are running on.
import numpy as np #for Python's array format -- numpy
from pathlib import Path #this is another way of dealing with paths which was chosen for convenience.
import pandas as pd #pandas dataframes are essentially excel for python, and dealing with this is the focus here.

To play with how to write functions, and to clean up load_ADNI, finding the file path is placed in a function. This is very specific and not flexible -- more generally useful ways of doing this will be better in future, but the function can be amended to reflect that. 

In [14]:
def find_data_path():
    '''
    Returns path of the data
    '''
    here_dir    = os.path.dirname(os.path.realpath('__file__'))
    par_dir = os.path.abspath(os.path.join(here_dir, os.pardir))
    dataset_dir = os.path.join(par_dir, 'data')
    return dataset_dir

In [15]:
find_data_path()

'/Users/Megan/RajLab/ADNI/data'

The next function is called load_baselines_into_dataframes(). Let's break down what it is doing step by step and explore the data in the process. 

In [19]:
dataset_dir = find_data_path()
print(dataset_dir)
adni = [dataset_dir + '/vec_a2b_1y_atr_factor.mat',
            dataset_dir + '/vec_a2b_3y_atr_factor.mat',
            dataset_dir + '/vec_a2b_5y_atr_factor.mat']

/Users/Megan/RajLab/ADNI/data


In [20]:
adni 

['/Users/Megan/RajLab/ADNI/data/vec_a2b_1y_atr_factor.mat',
 '/Users/Megan/RajLab/ADNI/data/vec_a2b_3y_atr_factor.mat',
 '/Users/Megan/RajLab/ADNI/data/vec_a2b_5y_atr_factor.mat']

In [21]:
adni = [loadmat(adni[0]),loadmat(adni[1]),loadmat(adni[2])]

Let's take a look at what this data looks like. It is a list of dictionaries. 

In [28]:
adni

[{'__globals__': [],
  '__header__': b'MATLAB 5.0 MAT-file, Platform: MACI64, Created on: Wed Oct 17 13:12:32 2018',
  '__version__': '1.0',
  'vec_a2b_1y_atr_factor': array([[3.00000e+00, 3.00000e+00, 3.00000e+00, ..., 1.90382e+06,
          0.00000e+00, 1.20000e+01],
         [3.00000e+00, 3.00000e+00, 3.00000e+00, ..., 1.90342e+06,
          0.00000e+00, 1.20000e+01],
         [4.00000e+00, 2.00000e+00, 2.00000e+00, ..., 1.65674e+06,
          0.00000e+00, 1.20000e+01],
         ...,
         [5.14200e+03, 1.00000e+00, 1.00000e+00, ..., 1.45484e+06,
          0.00000e+00, 1.20000e+01],
         [5.14200e+03, 1.00000e+00, 1.00000e+00, ..., 1.42988e+06,
          0.00000e+00, 1.20000e+01],
         [5.14700e+03, 1.00000e+00, 1.00000e+00, ..., 1.54621e+06,
          0.00000e+00, 1.20000e+01]])},
 {'__globals__': [],
  '__header__': b'MATLAB 5.0 MAT-file, Platform: MACI64, Created on: Wed Oct 17 13:13:14 2018',
  '__version__': '1.0',
  'vec_a2b_3y_atr_factor': array([[4.00000e+00, 2.00

Extract the year 1 data, addressing the actual data in the dictionary using the key 'vec_a2b_1y_atr_factor'

In [31]:
adni[0]['vec_a2b_1y_atr_factor'].shape

(1865, 207)

Get rid of metadata and rearrange in to a big list of matrices.

In [32]:
adni = [adni[0]['vec_a2b_1y_atr_factor'],adni[1]['vec_a2b_3y_atr_factor'],adni[2]['vec_a2b_5y_atr_factor']]

In [34]:
adni[0].shape

(1865, 207)

In [35]:
len(adni[0])

1865

So, we can address each year by the first index. There are 207 attributes, including 86-long atrophy vectors at baseline and the time point plus all sorts of other patient information. For instance, the number of months since the baseline measurement was taken. 

eg. make a list of the IDS of the patients for the year 1 data.

In [60]:
IDS = [adni[0][i][0] for i in range(len(adni[0]))]

In [61]:
IDS

[3.0,
 3.0,
 4.0,
 4.0,
 5.0,
 6.0,
 6.0,
 10.0,
 14.0,
 14.0,
 14.0,
 15.0,
 16.0,
 16.0,
 16.0,
 19.0,
 21.0,
 21.0,
 21.0,
 21.0,
 21.0,
 23.0,
 23.0,
 23.0,
 23.0,
 23.0,
 23.0,
 23.0,
 23.0,
 31.0,
 31.0,
 31.0,
 31.0,
 38.0,
 40.0,
 40.0,
 40.0,
 42.0,
 42.0,
 42.0,
 42.0,
 43.0,
 47.0,
 51.0,
 51.0,
 51.0,
 51.0,
 51.0,
 54.0,
 54.0,
 56.0,
 56.0,
 56.0,
 59.0,
 59.0,
 59.0,
 60.0,
 60.0,
 61.0,
 61.0,
 61.0,
 66.0,
 67.0,
 67.0,
 69.0,
 69.0,
 70.0,
 72.0,
 72.0,
 72.0,
 72.0,
 76.0,
 77.0,
 80.0,
 81.0,
 81.0,
 83.0,
 83.0,
 84.0,
 87.0,
 87.0,
 89.0,
 89.0,
 90.0,
 90.0,
 90.0,
 91.0,
 91.0,
 94.0,
 94.0,
 95.0,
 96.0,
 97.0,
 97.0,
 98.0,
 106.0,
 107.0,
 108.0,
 108.0,
 109.0,
 109.0,
 110.0,
 111.0,
 112.0,
 112.0,
 112.0,
 112.0,
 116.0,
 116.0,
 116.0,
 120.0,
 120.0,
 120.0,
 120.0,
 120.0,
 120.0,
 120.0,
 123.0,
 125.0,
 125.0,
 125.0,
 125.0,
 126.0,
 126.0,
 126.0,
 126.0,
 126.0,
 126.0,
 126.0,
 126.0,
 127.0,
 127.0,
 127.0,
 127.0,
 127.0,
 128.0,
 128.0,
 128.0

There are some repetitions, something that will be explored once we put it in to dataframes.

So, the bit of the code that does that. It loops over an iterator 'j' which addresses each of the three years, and for each feature loops through the full length of all the patient IDS. This actually results in three dataframes, one for each year, in a list. 

In [62]:
adni_df = [pd.DataFrame({
                        'ID'                        : [int(adni[j][i][0]) for i in range(len(adni[j]))],
                        'Baseline Dx'               : [int(adni[j][i][1]) for i in range(len(adni[j]))],
                        '1y Dx'                     : [int(adni[j][i][2]) for i in range(len(adni[j]))],
                        'End-of-study Dx'           : [int(adni[j][i][3]) for i in range(len(adni[j]))],
                        'Baseline atrophy'          : [adni[j][i][4:90] for i in range(len(adni[j]))],
                        'Future atrophy'            : [adni[j][i][90:176] for i in range(len(adni[j]))],
                        'Genetic Info (APOE4)'      : [adni[j][i][176] for i in range(len(adni[j]))],
                        'bl Age'                    : [adni[j][i][177] for i in range(len(adni[j]))],
                        'gender'                    : [adni[j][i][178] for i in range(len(adni[j]))],
                        'education year'            : [adni[j][i][179] for i in range(len(adni[j]))],
                        'marriage'                  : [adni[j][i][180] for i in range(len(adni[j]))],
                        'bl ADAS11'                 : [adni[j][i][181] for i in range(len(adni[j]))],
                        'bl ADAS13'                 : [adni[j][i][182] for i in range(len(adni[j]))],
                        'bl RAVLT_immediate'        : [adni[j][i][183] for i in range(len(adni[j]))],
                        'bl RAVLT_learning'         : [adni[j][i][184] for i in range(len(adni[j]))],
                        'bl RAVLT_forgetting'       : [adni[j][i][185] for i in range(len(adni[j]))],
                        'bl RAVLT_perc_forgetting'  : [adni[j][i][186] for i in range(len(adni[j]))],
                        'bl FAQ'                    : [adni[j][i][187] for i in range(len(adni[j]))],
                        'bl CDR'                    : [adni[j][i][188] for i in range(len(adni[j]))],
                        'bl MMSE'                   : [adni[j][i][189] for i in range(len(adni[j]))],
                        'ft Age'                    : [adni[j][i][190] for i in range(len(adni[j]))],
                        'gender'                    : [adni[j][i][191] for i in range(len(adni[j]))],
                        'education year'            : [adni[j][i][192] for i in range(len(adni[j]))],
                        'marriage'                  : [adni[j][i][193] for i in range(len(adni[j]))],
                        'ft ADAS11'                 : [adni[j][i][194] for i in range(len(adni[j]))],
                        'ft ADAS13'                 : [adni[j][i][195] for i in range(len(adni[j]))],
                        'ft RAVLT_immediate'        : [adni[j][i][196] for i in range(len(adni[j]))],
                        'ft RAVLT_learning'         : [adni[j][i][197] for i in range(len(adni[j]))],
                        'ft RAVLT_forgetting'       : [adni[j][i][198] for i in range(len(adni[j]))],
                        'ft RAVLT_perc_forgetting'  : [adni[j][i][199] for i in range(len(adni[j]))],
                        'ft FAQ'                    : [adni[j][i][200] for i in range(len(adni[j]))],
                        'ft CDR'                    : [adni[j][i][201] for i in range(len(adni[j]))],
                        'ft MMSE'                   : [adni[j][i][202] for i in range(len(adni[j]))],
                        'bl Intracranial Volume'    : [adni[j][i][203] for i in range(len(adni[j]))],
                        'ft Intracranial Volume'    : [adni[j][i][204] for i in range(len(adni[j]))],
                        'end-of-study is_convert'   : [adni[j][i][205] for i in range(len(adni[j]))],
                        'number of months from bl'  : [adni[j][i][206] for i in range(len(adni[j]))]
                         }) for j in range(3)]

Now we can start using Pandas functionality (I suppose we ought to know how to do this).

In [67]:
adni_df[0].head() #just display thte first few entries.

ID  Baseline Dx  1y Dx  End-of-study Dx  \
0   3            3      3                3   
1   3            3      3                3   
2   4            2      2                2   
3   4            2      2                2   
4   5            1      1                1   

                                    Baseline atrophy  \
0  [0.42899300267558316, 0.6101146248911582, 0.82...   
1  [0.49130654601460605, 0.5657404677305382, 0.76...   
2  [0.38735705827325556, 0.16893355463538812, 0.6...   
3  [0.3269262100101499, 0.27174938386729525, 0.68...   
4  [0.5883427925210888, 0.7011389219041197, 0.664...   

                                      Future atrophy  Genetic Info (APOE4)  \
0  [0.49130654601460605, 0.5657404677305382, 0.76...                   1.0   
1  [0.4797399004600559, 0.6013764392125581, 0.791...                   1.0   
2  [0.49331091696947693, 0.373494804762419, 0.685...                   0.0   
3  [0.540077225624267, 0.18117533851644893, 0.625...                   0.0   
4  [0.4331046075873159, 0.642863560168384, 0.6006...                   0.0   

     bl Age  gender  education year            ...             \
0  0.920354     1.0        0.570933            ...              
1  1.084444     1.0        0.570933            ...              
2 -1.345638     1.0       -2.351808            ...              
3 -1.266965     1.0       -2.351808            ...              
4  0.000596     1.0       -0.159752            ...              

   ft RAVLT_learning  ft RAVLT_forgetting  ft RAVLT_perc_forgetting  \
0          -1.634400             1.302757                  2.279192   
1          -2.054917             0.573340                  2.279192   
2          -0.793366            -0.885494                 -0.646280   
3           0.888701            -0.520786                 -0.506970   
4           0.047667             0.938048                  0.759468   

      ft FAQ     ft CDR   ft MMSE  bl Intracranial Volume  \
0  21.999117  12.774359 -9.680302               1920690.0   
1  18.069977  29.549822 -8.068754               1903820.0   
2  -0.266010   3.454657 -2.428334               1679440.0   
3   1.043703   3.454657 -1.622559               1661130.0   
4  -0.266010   3.454657  0.794764               1634180.0   

   ft Intracranial Volume  end-of-study is_convert  number of months from bl  
0               1903820.0                      0.0                      12.0  
1               1903420.0                      0.0                      12.0  
2               1656740.0                      0.0                      12.0  
3               1648510.0                      0.0                      12.0  
4               1632880.0                      0.0                      12.0  

[5 rows x 34 columns]

In [69]:
adni_df[0].describe() #not necessarily useful here, but a generally useful function.

ID  Baseline Dx        1y Dx  End-of-study Dx  \
count  1865.000000  1865.000000  1865.000000      1865.000000   
mean   1991.796247     1.812869     1.878820         2.067560   
std    1760.258624     0.665558     0.728266         0.814135   
min       3.000000     1.000000     1.000000         1.000000   
25%     522.000000     1.000000     1.000000         1.000000   
50%    1155.000000     2.000000     2.000000         2.000000   
75%    4173.000000     2.000000     2.000000         3.000000   
max    5147.000000     3.000000     3.000000         3.000000   

       Genetic Info (APOE4)       bl Age       gender  education year  \
count           1865.000000  1865.000000  1865.000000     1865.000000   
mean               0.557105    -0.315146     1.453619       -0.145452   
std                0.666556     1.120308     0.497978        1.013903   
min                0.000000    -3.398167     1.000000       -3.813178   
25%                0.000000    -1.049737     1.000000       -0.890438   
50%                0.000000    -0.261903     1.000000       -0.159752   
75%                1.000000     0.444167     2.000000        0.570933   
max                2.000000     2.777301     2.000000        1.301618   

          marriage    bl ADAS11            ...             ft RAVLT_learning  \
count  1865.000000  1865.000000            ...                   1865.000000   
mean      2.286863     1.585068            ...                     -0.655937   
std       0.701595     2.242841            ...                      1.179793   
min       0.000000    -1.937757            ...                     -3.316467   
25%       2.000000    -0.099709            ...                     -1.634400   
50%       2.000000     1.003120            ...                     -0.793366   
75%       2.000000     2.719856            ...                      0.047667   
max       4.000000    12.031405            ...                      3.411802   

       ft RAVLT_forgetting  ft RAVLT_perc_forgetting       ft FAQ  \
count          1865.000000               1865.000000  1865.000000   
mean              0.293209                  0.862821     6.139998   
std               0.942934                  1.280223     9.428528   
min              -5.626705                -14.437788    -0.266010   
25%              -0.520786                 -0.228355    -0.266010   
50%               0.208631                  0.846309     1.043703   
75%               0.938048                  2.279192    10.211697   
max               4.220425                  2.279192    39.025391   

            ft CDR      ft MMSE  bl Intracranial Volume  \
count  1865.000000  1865.000000            1.865000e+03   
mean      6.665832    -1.683046            1.529524e+06   
std       8.647229     2.908585            1.598744e+05   
min      -0.273224   -16.932271            7.161330e+05   
25%      -0.273224    -2.428334            1.419560e+06   
50%       3.454657    -0.816785            1.519400e+06   
75%      10.910419     0.794764            1.632775e+06   
max      51.917107     0.794764            2.110290e+06   

       ft Intracranial Volume  end-of-study is_convert  \
count            1.865000e+03              1865.000000   
mean             1.528636e+06                 0.083110   
std              1.605601e+05                 0.276122   
min              7.161330e+05                 0.000000   
25%              1.418240e+06                 0.000000   
50%              1.518680e+06                 0.000000   
75%              1.632270e+06                 0.000000   
max              2.108790e+06                 1.000000   

       number of months from bl  
count                    1865.0  
mean                       12.0  
std                         0.0  
min                        12.0  
25%                        12.0  
50%                        12.0  
75%                        12.0  
max                        12.0  

[8 rows x 32 columns]

In [70]:
adni_df[0].info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1865 entries, 0 to 1864
Data columns (total 34 columns):
ID                          1865 non-null int64
Baseline Dx                 1865 non-null int64
1y Dx                       1865 non-null int64
End-of-study Dx             1865 non-null int64
Baseline atrophy            1865 non-null object
Future atrophy              1865 non-null object
Genetic Info (APOE4)        1865 non-null float64
bl Age                      1865 non-null float64
gender                      1865 non-null float64
education year              1865 non-null float64
marriage                    1865 non-null float64
bl ADAS11                   1865 non-null float64
bl ADAS13                   1865 non-null float64
bl RAVLT_immediate          1865 non-null float64
bl RAVLT_learning           1865 non-null float64
bl RAVLT_forgetting         1865 non-null float64
bl RAVLT_perc_forgetting    1865 non-null float64
bl FAQ                      1865 non-null float64
bl 

So there are no missing rows of data, which is good!

## Aggregation

Without too much reference to what the data is actually showing us yet, look at how to manipulate data frames.

How many different patients do we have?

In [74]:
adni_df[0]['ID'].nunique()

914

It would be nice to be able to aggregate data for individual patient IDs. While this doesn't make sense all the time (eg. when a single ID has different values on repeat -- why?) we can still look at some of the aspects of the data this way. 

To do this, we need an aggregation rule for each column. This is a method for returning the column names, and then converting to a list. 

In [83]:
adni_df[0].columns.values.tolist()

['ID',
 'Baseline Dx',
 '1y Dx',
 'End-of-study Dx',
 'Baseline atrophy',
 'Future atrophy',
 'Genetic Info (APOE4)',
 'bl Age',
 'gender',
 'education year',
 'marriage',
 'bl ADAS11',
 'bl ADAS13',
 'bl RAVLT_immediate',
 'bl RAVLT_learning',
 'bl RAVLT_forgetting',
 'bl RAVLT_perc_forgetting',
 'bl FAQ',
 'bl CDR',
 'bl MMSE',
 'ft Age',
 'ft ADAS11',
 'ft ADAS13',
 'ft RAVLT_immediate',
 'ft RAVLT_learning',
 'ft RAVLT_forgetting',
 'ft RAVLT_perc_forgetting',
 'ft FAQ',
 'ft CDR',
 'ft MMSE',
 'bl Intracranial Volume',
 'ft Intracranial Volume',
 'end-of-study is_convert',
 'number of months from bl']

We want to make a dictionary of the functions to apply for aggregation.

In [85]:
aggfunctions = dict((el,0) for el in adni_df[0].columns.values.tolist())
print(aggfunctions)

{'ID': 0, 'Baseline Dx': 0, '1y Dx': 0, 'End-of-study Dx': 0, 'Baseline atrophy': 0, 'Future atrophy': 0, 'Genetic Info (APOE4)': 0, 'bl Age': 0, 'gender': 0, 'education year': 0, 'marriage': 0, 'bl ADAS11': 0, 'bl ADAS13': 0, 'bl RAVLT_immediate': 0, 'bl RAVLT_learning': 0, 'bl RAVLT_forgetting': 0, 'bl RAVLT_perc_forgetting': 0, 'bl FAQ': 0, 'bl CDR': 0, 'bl MMSE': 0, 'ft Age': 0, 'ft ADAS11': 0, 'ft ADAS13': 0, 'ft RAVLT_immediate': 0, 'ft RAVLT_learning': 0, 'ft RAVLT_forgetting': 0, 'ft RAVLT_perc_forgetting': 0, 'ft FAQ': 0, 'ft CDR': 0, 'ft MMSE': 0, 'bl Intracranial Volume': 0, 'ft Intracranial Volume': 0, 'end-of-study is_convert': 0, 'number of months from bl': 0}


To make our lives easy, for now we'll just take the first for every ID. 

An example, where just the baseline Dx data is aggregated.

In [96]:
adni_df[0][['ID','Baseline Dx']].groupby(adni_df[0]['ID']).aggregate('first')

ID  Baseline Dx
ID                     
3        3            3
4        4            2
5        5            1
6        6            2
10      10            3
14      14            1
15      15            1
16      16            1
19      19            1
21      21            1
23      23            1
31      31            1
38      38            2
40      40            1
42      42            2
43      43            1
47      47            1
51      51            2
54      54            2
56      56            1
59      59            1
60      60            2
61      61            1
66      66            1
67      67            1
69      69            1
70      70            1
72      72            1
76      76            3
77      77            2
...    ...          ...
5007  5007            2
5012  5012            3
5014  5014            2
5015  5015            3
5019  5019            3
5023  5023            1
5026  5026            2
5027  5027            3
5028  5028            3
5040  5040            1
5054  5054            3
5057  5057            3
5063  5063            3
5078  5078            1
5079  5079            1
5082  5082            1
5083  5083            1
5095  5095            3
5096  5096            2
5097  5097            1
5100  5100            1
5118  5118            1
5125  5125            1
5127  5127            1
5129  5129            1
5130  5130            1
5132  5132            1
5137  5137            1
5142  5142            1
5147  5147            1

[914 rows x 2 columns]

In [98]:
aggfunctions = dict((el,'first') for el in adni_df[0].columns.values.tolist())
print(aggfunctions)

{'ID': 'first', 'Baseline Dx': 'first', '1y Dx': 'first', 'End-of-study Dx': 'first', 'Baseline atrophy': 'first', 'Future atrophy': 'first', 'Genetic Info (APOE4)': 'first', 'bl Age': 'first', 'gender': 'first', 'education year': 'first', 'marriage': 'first', 'bl ADAS11': 'first', 'bl ADAS13': 'first', 'bl RAVLT_immediate': 'first', 'bl RAVLT_learning': 'first', 'bl RAVLT_forgetting': 'first', 'bl RAVLT_perc_forgetting': 'first', 'bl FAQ': 'first', 'bl CDR': 'first', 'bl MMSE': 'first', 'ft Age': 'first', 'ft ADAS11': 'first', 'ft ADAS13': 'first', 'ft RAVLT_immediate': 'first', 'ft RAVLT_learning': 'first', 'ft RAVLT_forgetting': 'first', 'ft RAVLT_perc_forgetting': 'first', 'ft FAQ': 'first', 'ft CDR': 'first', 'ft MMSE': 'first', 'bl Intracranial Volume': 'first', 'ft Intracranial Volume': 'first', 'end-of-study is_convert': 'first', 'number of months from bl': 'first'}


However, trying *new_adni_df = adni_df[0].groupby(adni_df[0]['ID']).aggregate(aggfunctions)*
throws up all sorts of errors, namely that it doesn't aggregate (helpful!).

In [102]:
baseline_agg = adni_df[0][['ID','Baseline atrophy']].groupby(adni_df[0]['ID']).aggregate('first')

In [104]:
new_adni_df = adni_df[0].groupby(adni_df[0]['ID']).aggregate('first')

In [105]:
new_adni_df.head()

Baseline Dx  1y Dx  End-of-study Dx  \
ID                                        
3             3      3                3   
4             2      2                2   
5             1      1                1   
6             2      2                2   
10            3      3                3   

                                     Baseline atrophy  \
ID                                                      
3   [0.42899300267558316, 0.6101146248911582, 0.82...   
4   [0.38735705827325556, 0.16893355463538812, 0.6...   
5   [0.5883427925210888, 0.7011389219041197, 0.664...   
6   [0.15479104616439587, 0.4680471723600093, 0.60...   
10  [0.3517004675723798, 0.573498452226188, 0.8714...   

                                       Future atrophy  Genetic Info (APOE4)  \
ID                                                                            
3   [0.49130654601460605, 0.5657404677305382, 0.76...                   1.0   
4   [0.49331091696947693, 0.373494804762419, 0.685...                   0.0   
5   [0.4331046075873159, 0.642863560168384, 0.6006...                   0.0   
6   [0.09668576489342542, 0.4138680271876763, 0.54...                   0.0   
10  [0.5556972838750345, 0.5752348951844986, 0.895...                   1.0   

      bl Age  gender  education year  marriage            ...             \
ID                                                        ...              
3   0.920354     1.0        0.570933       2.0            ...              
4  -1.345638     1.0       -2.351808       2.0            ...              
5   0.000596     1.0       -0.159752       2.0            ...              
6   0.772572     2.0       -1.255780       2.0            ...              
10 -0.131103     2.0       -1.621123       2.0            ...              

    ft RAVLT_learning  ft RAVLT_forgetting  ft RAVLT_perc_forgetting  \
ID                                                                     
3           -1.634400             1.302757                  2.279192   
4           -0.793366            -0.885494                 -0.646280   
5            0.047667             0.938048                  0.759468   
6           -0.793366             1.667466                  2.279192   
10          -2.475434            -0.156077                  2.279192   

       ft FAQ     ft CDR   ft MMSE  bl Intracranial Volume  \
ID                                                           
3   21.999117  12.774359 -9.680302               1920690.0   
4   -0.266010   3.454657 -2.428334               1679440.0   
5   -0.266010   3.454657  0.794764               1634180.0   
6   -0.266010   5.318598 -2.428334               1485830.0   
10  25.928257  18.366180 -0.816785               1467270.0   

    ft Intracranial Volume  end-of-study is_convert  number of months from bl  
ID                                                                             
3                1903820.0                      0.0                      12.0  
4                1656740.0                      0.0                      12.0  
5                1632880.0                      0.0                      12.0  
6                1512740.0                      0.0                      12.0  
10               1469530.0                      0.0                      12.0  

[5 rows x 33 columns]

In [113]:
len(adni_df[0].columns.values.tolist())

34

Note, the act of making a dataframe has removed duplicate representations of the same data -- gender, education and marriage were all repeated in the matrices, and have just been written once to the dataframe.

We can count the different contributions of some of the columns. For instance:

In [122]:
new_adni_df['end-of-study is_convert'].value_counts()

0.0    856
1.0     58
Name: end-of-study is_convert, dtype: int64

In [123]:
new_adni_df['marriage'].value_counts()

2.0    699
4.0     98
3.0     84
1.0     29
0.0      4
Name: marriage, dtype: int64

In [124]:
new_adni_df['gender'].value_counts()

1.0    489
2.0    425
Name: gender, dtype: int64

## Trying to learn something about the diagnoses